### PR TITLE
feature(receive): Receive new design funnel

### PR DESCRIPTION
**Pull request by euvini**

### CODE REVIEW

**Review by SorinC6 — COMMENTED** [review]
Please check again for example how the feature `RampOnOff` or `Swap` is structured as will need to make this changes: 
- the new design for this feature must be added inside `src/features` (see Swap as example)
- the feature folder must have two folders : `common` and `useCases`.  
  `common` folder will have: 
      -  the actual componets that are used for this feature  
      - all the `useStrings/strings` used for this feature 
      - all the navigation : `useNavigateTo/navigation` used for this feature
      -all the mocks/helpers/types, see more examples in the features mentioned above.

   `useCases` will have 
      - all the screens for this feature, see feature `RampOnOff` or `Swap` on how to structure the screen 

- At the root of the feature can add a `ReceiveNavigator` 

- the styles are not added properly - can bring the latest changes on develop as I see that is not up to date with it and then can see more exampes how to properly add the styles as the other features provided. See guide here https://emurgo.atlassian.net/wiki/x/BQBrPQ 

- please check how the ilustration are added in the features provided as examples 

Basically please check how `RampOnOff` or `Swap` feature are structured

File: apps/wallet-mobile/src/Receive/AddressDetail.tsx [hunk 1]
```diff
@@ -1,25 +1,322 @@
...
+      try {
+        const uri = await captureRef(ref, {
+          format: 'png',
+          quality: 1,
+          fileName: 'Share address'
+        });
+        setIsSharing(false);
+        await Share.open({ url: uri, filename: 'Share address', message: `${strings.address} ${address}` });
+      } catch (e) {
+        console.log(e);
```
Comment (jorbuedo) [inline apps/wallet-mobile/src/Receive/AddressDetail.tsx:142]: Careful with the logs. Is there's anything better to do here? Otherwise remove it when ready to be merged.

File: apps/wallet-mobile/src/Receive/AddressDetail.tsx [hunk 2]
```diff
@@ -1,25 +1,322 @@
...
-  </View>
+export function ShareQRCodeCard({ address, title }: ShareProps) {
+  const [isSharing, setIsSharing] = useState<boolean>(false);
+  const strings = useStrings()
+  const ref = useRef<any>();
+
+  const shareImage = () => {
+    setIsSharing(true);
+
+    setTimeout(async () => {
```
Comment (jorbuedo) [inline apps/wallet-mobile/src/Receive/AddressDetail.tsx:132]: Looks like a timeout to fix something that shouldn't be fixed by a timeout.

File: apps/wallet-mobile/src/Receive/ReceiveScreen.tsx [hunk 3]
```diff
@@ -1,30 +1,31 @@
-import {useFocusEffect} from '@react-navigation/native'
+import { useFocusEffect } from '@react-navigation/native'
```
Comment (jorbuedo) [inline apps/wallet-mobile/src/Receive/ReceiveScreen.tsx:1]: Formatting issues?

File: apps/wallet-mobile/src/Receive/AddressDetail.tsx [hunk 4]
```diff
@@ -1,25 +1,322 @@
...
+  const circleIndex = Array.from({ length: totalPages }, (_, index) => index + 1);
+
+  const onPageChange = (event) => {
+    const offset = event.nativeEvent.contentOffset.x;
+    const index = Math.floor(offset / (itemsPerPage * SCREEN_WIDTH - 64));
+    setScrollPosition(index);
+  };
+
+  return (
+    <>
```
Comment (banklesss) [inline apps/wallet-mobile/src/Receive/AddressDetail.tsx:45]: If its the screen root, it needs `SafeAreaView` from `react-native-safe-area-context` with the proper edges.

File: apps/wallet-mobile/src/Receive/AddressDetail.tsx [hunk 5]
```diff
@@ -1,25 +1,322 @@
...
+    </View>
+  )
 }
 
-export const AddressDetail = ({address}: Props) => (
-  <View style={styles.container}>
-    <QRCode value={address} size={140} backgroundColor={COLORS.LIGHT_GRAY} color="black" />
-  </View>
+export function ShareQRCodeCard({ address, title }: ShareProps) {
+  const [isSharing, setIsSharing] = useState<boolean>(false);
```
Comment (banklesss) [inline apps/wallet-mobile/src/Receive/AddressDetail.tsx:125]:
```suggestion
  const [isSharing, setIsSharing] = useState(false);
```

File: apps/wallet-mobile/src/Receive/AddressDetail.tsx [hunk 6]
```diff
@@ -1,25 +1,322 @@
...
+  const strings = useStrings()
+  const ref = useRef<any>();
```
Comment (banklesss) [inline apps/wallet-mobile/src/Receive/AddressDetail.tsx:127]: Please define the ref type shape.

File: apps/wallet-mobile/src/Receive/AddressDetail.tsx (hunk 2 above)
Comment (banklesss) [reply apps/wallet-mobile/src/Receive/AddressDetail.tsx:132]: maybe `react-query` can do the job.

File: apps/wallet-mobile/src/Receive/AddressDetail.tsx [hunk 7]
```diff
@@ -1,25 +1,322 @@
...
+      style={[styles.touchableCard]}
+    >
+      <LinearGradient
+        style={[StyleSheet.absoluteFill, { opacity: 1 }]}
+        start={{ x: 0, y: 0 }}
+        end={{ x: 0, y: 1 }}
+        colors={['#E4E8F7', '#C6F7F7']}
+      />
+
+      <Image ref={LogoRef.current} source={require('../assets/img/yoroi-logo-text.png')} style={{ height: 40, width: 40 }} />
```
Comment (banklesss) [inline apps/wallet-mobile/src/Receive/AddressDetail.tsx:111]: We avoid png images. Could you use the svg and transform it to react native component using this tool: https://transform.tools/svg-to-react-native

File: apps/wallet-mobile/src/Receive/ReceiveScreen.tsx [hunk 8]
```diff
@@ -41,63 +42,57 @@ export const ReceiveScreen = () => {
...
-          <Spacer height={24} />
+          <Button
+            shelleyTheme
+            onPress={() => {
+              copy(currentAddress)
+            }}
+            disabled={currentAddress === null ? true : false}
+            testID="copyReceiveAddressButton"
+            title='Copy address'
+            iconImage={require('../../src/assets/img/copy.png')}
```
Comment (banklesss) [inline apps/wallet-mobile/src/Receive/ReceiveScreen.tsx:72]: We avoid png images. Could you use the svg and transform it to react native component using this tool: https://transform.tools/svg-to-react-native

File: apps/wallet-mobile/src/features/Receive/common/CaptureShareQRCodeCard/CaptureShareQRCodeCard.tsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+            style={[styles.touchableCard]}
+        >
+            <LinearGradient
+                style={[StyleSheet.absoluteFill, { opacity: 1 }]}
+                start={{ x: 0, y: 0 }}
+                end={{ x: 0, y: 1 }}
+                colors={['#E4E8F7', '#C6F7F7']}
+            />
+
+            <Image ref={LogoRef.current} source={require('../assets/img/yoroi-logo-text.png')} style={{ height: 40, width: 40 }} />
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Receive/common/CaptureShareQRCodeCard/CaptureShareQRCodeCard.tsx:35]: We avoid png images. Could you use the svg and transform it to react native component using this tool: https://transform.tools/svg-to-react-native

File: apps/wallet-mobile/src/features/Receive/useCases/ReceiveScreen.tsx [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+                    <Button
+                        shelleyTheme
+                        onPress={() => {
+                            copy(currentAddress!)
+                        }}
+                        disabled={currentAddress === null ? true : false}
+                        testID="copyReceiveAddressButton"
+                        title='Copy address'
+                        iconImage={require('../../src/assets/img/copy.png')}
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Receive/useCases/ReceiveScreen.tsx:75]: We avoid png images. Could you use the svg and transform it to react native component using this tool: https://transform.tools/svg-to-react-native

File: apps/wallet-mobile/src/features/Receive/common/ShareQRCodeCard/ShareQRCodeCard.tsx [hunk 11]
```diff
@@ -0,0 +1,134 @@
+import React, { RefObject, useRef, useState } from 'react'
+import { Alert, StyleSheet, TouchableOpacity, View } from 'react-native'
+import QRCode from 'react-native-qrcode-svg'
+import Share from 'react-native-share';
+import ViewShot, { captureRef } from "react-native-view-shot";
+import { Text } from 'src/components';
+import { COLORS } from 'src/theme';
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Receive/common/ShareQRCodeCard/ShareQRCodeCard.tsx:7]: It needs a relative path

File: apps/wallet-mobile/src/features/Receive/common/ShareDetailsCard/ShareDetailsCard.tsx [hunk 12]
```diff
@@ -0,0 +1,77 @@
+import React, {  } from 'react'
+import { StyleSheet, View } from 'react-native'
+import { CopyButton, Text } from 'src/components';
+import { COLORS } from 'src/theme';
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Receive/common/ShareDetailsCard/ShareDetailsCard.tsx:4]: It needs a relative path

File: apps/wallet-mobile/src/features/Receive/common/CaptureShareQRCodeCard/CaptureShareQRCodeCard.tsx [hunk 13]
```diff
@@ -0,0 +1,70 @@
+import React, { useRef } from 'react'
+import { Image, StyleSheet, View } from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import QRCode from 'react-native-qrcode-svg'
+import { Text } from 'src/components'
+import { COLORS } from 'src/theme'
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Receive/common/CaptureShareQRCodeCard/CaptureShareQRCodeCard.tsx:6]: It needs a relative path

File: apps/wallet-mobile/src/features/Receive/common/AddressDetail/AddressDetail.tsx [hunk 14]
```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react'
+import { Dimensions, FlatList, StyleSheet, View } from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import { COLORS, colors } from 'src/theme';
```
Comment (banklesss) [inline apps/wallet-mobile/src/features/Receive/common/AddressDetail/AddressDetail.tsx:4]: It needs a relative path
